File: 04122024_need_for_speed/fibonacci_sequence.c
// C code to find the nth number of the Fibonnaci sequence
#include <stdio.h>
#include <time.h>

int fibo(int n) {
    if (n <= 1)
        return n;
    else
        return fibo(n - 1) + fibo(n - 2);
}

int main() {
    int num = 40;
    clock_t start_time, end_time;
    double cpu_time_used;

    start_time = clock();

    printf("Fibonacci Number %d\n", fibo(num));

    end_time = clock();
    cpu_time_used = ((double) (end_time - start_time)) / CLOCKS_PER_SEC;
    printf("--- %f seconds ---\n", cpu_time_used);

    return 0;
}

File: 04122024_need_for_speed/fibonacci_sequence.py
# python code to find the nth fibonacci number
import time

def fibonacci(n):
    if n <= 1:
        return n
    else:
        return(fibonacci(n-1) + fibonacci(n-2))

num = 40

# let's find the 50th fibonacci number
start_time = time.time()

fibonacci(num)

print("Time Taken: %s seconds" % (time.time() - start_time))